storage/storage-client/src/lib.rs

```diff
@@ -53,22 +52,15 @@ impl StorageClient {
         bcs::from_bytes(&result)?
     }
 
-    pub fn get_account_state_with_proof_by_version(
+    pub fn get_value_with_proof_by_version(
```

Comment:
outside of the state_db, a `value` does sound ambiguous, let's use `state_value`  (or `state_item` but you didn't use `item` anywhere, so forget it :) )